pkg/cmd/releasenotes/render.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (C) 2021 ScyllaDB
+
+package releasenotes
+
+import (
+	"fmt"
+	"html/template"
+	"io"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+const releaseNotesTemplate = `
+# Release notes for {{ .Release }}
+## Container images
+
+` + "```\ndocker.io/scylladb/scylla-operator:{{ .Release }}\n```" + `
+
+## Changes By Kind
+
+{{- range $section := .Sections }}
+  {{- if $section.PullRequests }}
+
+### {{ $section.Title }}
+
+    {{- range $pr := $section.PullRequests }}
+{{ FormatPR $pr }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+---
+`
+
+var branchPrefixRe = regexp.MustCompile(`^(\[[^]]+])?(.*)`)
+
+func cleanupPRTitle(title string) (string, error) {
+	m := branchPrefixRe.FindStringSubmatch(strings.TrimSpace(title))
+	if len(m) != 3 {
+		return "", fmt.Errorf("can't parse PR title %q", title)
+	}
+	return strings.TrimSpace(m[2]), nil
+}
+
+func prURL(pr PullRequest) string {
+	return fmt.Sprintf("https://github.com/%s/%s/pull/%d", pr.Repository.Owner.Login, pr.Repository.Name, pr.Number)
+}
+
+func userURL(login string) string {
+	return fmt.Sprintf("https://github.com/%s", login)
+}
+
+func formatPR(pr PullRequest) (string, error) {
+	title, err := cleanupPRTitle(string(pr.Title))
+	if err != nil {
+		return "", err
+	}
+	userLogin := pr.Author.Login
+	details := fmt.Sprintf("[#%d](%s),[@%s](%s)", pr.Number, prURL(pr), userLogin, userURL(string(userLogin)))
+	return fmt.Sprintf("* %s (%s)", title, details), nil
+}
+
+type section struct {
+	Title        string
+	PullRequests []PullRequest
+}
+
+type releaseNoteData struct {
+	Sections []section
+	Release  string
+}
+
+func renderReleaseNotes(out io.Writer, release string, pullRequests []PullRequest) error {
+	// Render in the order of merge date.
+	sort.Slice(pullRequests, func(i, j int) bool {
+		return pullRequests[i].MergedAt.Before(pullRequests[j].MergedAt.Time)
+	})
+
+	prsByKind, uncategorized := splitByKind(pullRequests)
+
+	sections := []section{
+		{"API Changes", extractByKind(prsByKind, []string{"api-change"})},
+		{"Features", extractByKind(prsByKind, []string{"feature"})},
+		{"Bugs", extractByKind(prsByKind, []string{"bug"})},
+		{"Documentation", extractByKind(prsByKind, []string{"documentation"})},
+		{"Flaky/Failing Tests", extractByKind(prsByKind, []string{"failing-test", "flake"})},
+		{"Other", flatten(prsByKind)},
```

Comment:
guess we should unify the singular/plural form. If we say *bugs* we should use *others* or the other way around